examples/src/main/kotlin/io/grpc/examples/routeguide/RouteGuideServer.kt

```diff
@@ -127,8 +125,8 @@ class RouteGuideServer private constructor(
       }.build()
     }
 
-    override fun routeChat(requests: Flow<RouteNote>): Flow<RouteNote> {
-      return flow {
+    override fun routeChat(requests: Flow<RouteNote>): Flow<RouteNote> =
```

Comment:
@lowasser - it is my impression that this implementation of `routeChat()` actually processes all client request messages before issuing a stream of responses. Is that correct, or am I misunderstanding the code?

Reply:
No, that's not correct.  Flows are a little nonintuitive at first, but what's happening is approximately that `collect` is setting up what amounts to a listener that processes requests as they arrive, and `emit` is immediately sending a response back to the client.